JunQueue.java

```diff
@@ -1,249 +1,65 @@
-                    garbageCollection();
-                }
-            
-            switch (command) {
-                case "done": // DONE
-                    done = true;
-                    break label;
-                    
-                case "specs": // DONE
-                    queueOperations();
-                    break;
-                    
-                case "enqueue": // DONE
-                    System.out.println("Input character to add to front of the Queue.");
-                    String value_to_add = userInput.next();
-                    
-                    enqueue(value_to_add);
-                    System.out.println(Arrays.toString(this.queue));
-                    break;
-                    
-                case "dequeue": // DONE
-                    dequeue();
-                    System.out.println(Arrays.toString(this.queue));
-                    break;
-                    
-                case "isempty": // DONE
-                    isEmpty();
-                    System.out.println(isEmpty());
-                    break;
-                    
-                case "isfull": // DONE
-                    isFull();
-                    System.out.println(isFull());
-                    break;
-                    
-                case "peek": // DONE
-                    peek(); 
-                    System.out.println(peek());
-                    break;
 
-                case "count": // DONE
-                    System.out.println(trackQueuePosition);
-                    break;
-                    
-                case "display": // DONE
-                    display(); 
-                    break;
-                
-                case "totaldequeue": // DONE
-                    totalDequeue();
-                    break;
-                
-                default:
-                    System.out.println("That is not an option. Please input 'specs' if you need clarification on what you can call.");
-                    System.out.println("Remember each command executes only one time.");
-                    break;
-            }
-        }
-    }
-    
-	//________________________________THIS IS THE START OF THE QUEUE METHODS__________________________________________
-    
-    public void garbageCollection(){
-        if (trackQueuePosition == 0){
-            queue = new String[1]; // prints null
-            queue[0] = null;
-        }
-        else{
-            String[] newqueue = new String[trackQueuePosition * 2];
-            System.arraycopy(queue, 0, newqueue, 0, trackQueuePosition);
-            this.queue = newqueue;
-            }
-        }
-    
     public void enqueue(String value) {
-        //creates list twice as big when necessary
-        if (value.length() != 1) {
-            System.out.println("Not valid input. Must be 1 character only.");

```

Comment:
Your list would count " " as a valid variable because it's only checking the length, not to see whether or not there is something actually in the string, this change isn't needed if you want that as intended behavior but just fyi.

Reply:
and isEmpty is more readable.

Reply:
might need to revisit this. confused here